service/controller/machinepoolexp/handler/migration/convert.go

```diff
@@ -0,0 +1,49 @@
+package migration
+
+import (
+	"encoding/json"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func cloneObjectMeta(objectMeta metav1.ObjectMeta) metav1.ObjectMeta {
+	return metav1.ObjectMeta{
+		Name:            objectMeta.Name,
+		Namespace:       objectMeta.Namespace,
+		Labels:          objectMeta.Labels,
+		Annotations:     objectMeta.Annotations,
+		OwnerReferences: objectMeta.OwnerReferences,
+		Finalizers:      objectMeta.Finalizers,
+		ClusterName:     objectMeta.ClusterName,
+	}
+}
+
+func convertSpec(oldSpec interface{}, newSpec interface{}) error {
```

Comment:
Copies CR Spec from old to new CR, by serializing the old object and deserializing in into the new one (old and new are different Go types, but they are identical).